### PR TITLE
VHeader

### LINKED ISSUE #462: Header
## Description

The header...it's pretty important! The first thing at the top of the page 😅 .

### Features

- Sticky (persists at the top of the page after the user scrolls)
- Has a bottom border when the filter sidebar is active _and_ after the user scrolls down the page

### Props

| Name | Type | Description |
|-|-|-|

### Slots

| Name | Slot props | Description |
|-|-|-|

### State

| Name | Type | Description |
|-|-|-|

### Events

| Name | Associated data | Description |
|-|-|-|

## Children

- [x] #448 
- [x] #375 
- [x] #453
- [x] #476
- [x] #485 

## References

Desktop: https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=751%3A2802
Mobile: https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=751%3A2808

## Implementation

- [x] 🙋 I would be interested in implementing this component.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #462 by @zackkrida

## Description

This is a feature branch that is the base for several Header components. Currently, this PR only contains a stub of the Header.

### Todo

- [ ] Fix filter button tests as suggested in https://github.com/WordPress/openverse-frontend/pull/489#discussion_r770696201

## Testing Instructions

None yet.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Noting that I've changed the base branch here to the `redesign` branch, so we don't block pushing things to `main` if we need to.

File: src/components/VHeader/VHeader.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<template>
+  <div
+    class="fixed flex px-8 align-center z-40 w-full bg-white"
```
Comment (zackkrida) [inline src/components/VHeader/VHeader.vue:3]:
```suggestion
    class="sticky top-0 flex px-8 align-center z-40 w-full bg-white"
```

This should be preferable for the header, then we can remove the padding from the 'embedded' layout and the 'border appearing scroll' behavior should be smoother.

**Comment by zackkrida** [comment]
Noting that this branch is currently broken due to a  `Cannot read properties of undefined (reading 'isFetching')` error which I believe will be fixed when #489 is merged.

**Comment by dhruvkb** [comment]
The `InputField` stories are broken by the file rename to `VInputField` (@ `src/components/VInputField/VInputField.vue`).

File: src/components/VInputField/VInputField.vue [hunk 2]
```diff
@@ -12,39 +12,31 @@
...
  * Provides a control to enter text as input.
  */
 export default {
-  name: 'InputField',
+  name: 'VInputField',
   inheritAttrs: false,
-  model: {
-    prop: 'value',
-    event: 'input',
-  },
```
Comment (dhruvkb) [inline src/components/VInputField/VInputField.vue:42]: Removing this prevents the field from working with `v-model`.

Comment (dhruvkb) [reply src/components/VInputField/VInputField.vue:42]: See #586

**Comment by zackkrida** [comment]
@WordPress/openverse-frontend I'm starting to think we should just merge this into `redesign` and use that as the base branch for all new changes. A lot of this work is interconnected because the header is such a crucial component. and there's changes here that would be beneficial to other PRs. What do you think?

**Comment by zackkrida** [comment]
@WordPress/openverse-frontend I'm going to merge this, I'll go ahead and assign a new base branch to each relevant PR. I think this can get us moving faster! 😄

**Pull request merged**